sourcery_analytics/extractors.py

```diff
@@ -159,6 +167,29 @@ def _extract_from_directory(self, directory: pathlib.Path) -> typing.Iterator[T]
         )
 
     def _extract_from_file(self, file: pathlib.Path) -> typing.Iterator[T]:
-        manager = astroid.manager.AstroidManager()
-        module = manager.ast_from_file(file)
-        yield from self._extract_from_node(module)
+        try:
+            module = self.manager.ast_from_file(file)
+            yield from self._extract_from_node(module)
+        except astroid.AstroidSyntaxError as e:
+            if isinstance(e.error, SyntaxError):
+                error_message = _format_syntax_error_message(
+                    "skipping file", file, e.error
+                )
+            else:
+                error_message = str(e).replace("\n", " ")
+            warning = SyntaxWarning(error_message)
+            warnings.warn(warning)
+            yield from ()
```

Comment:
**suggestion (code_refinement):** Ensure proper handling of syntax errors for robustness.

The addition of syntax error handling in `_extract_from_file` is a good improvement for robustness. It might be worth considering if there are any specific actions or logging that should occur when a syntax error is encountered, beyond just warning and skipping the file, to aid in debugging or further analysis.

```suggestion
        except astroid.AstroidSyntaxError as e:
            if isinstance(e.error, SyntaxError):
                error_message = _format_syntax_error_message(
                    "Syntax error encountered, skipping file", file, e.error
                )
                # Log the syntax error for further analysis
                logging.error(f"Syntax error in file {file}: {e.error}")
            else:
                error_message = str(e).replace("\n", " ")
                # Log generic astroid errors
                logging.error(f"Astroid error in file {file}: {error_message}")
            warning = SyntaxWarning(error_message)
            warnings.warn(warning)
            # Optionally, you could add more specific handling here, such as:
            # - Saving the file path and error details to a report for later analysis
            # - Attempting to auto-correct known syntax issues and reprocess the file
            # - Emitting metrics or alerts for monitoring the frequency of syntax errors
            yield from ()
```